readium/streamer/src/main/java/org/readium/r2/streamer/parser/epub/AccessibilityAdapter.kt

```diff
@@ -0,0 +1,168 @@
+                .toSet()
+                .takeUnless(Set<Accessibility.PrimaryAccessMode>::isEmpty)
+            }.toSet()
+        }
+
+    private fun adaptCertification(items: List<MetadataItem>): Pair<Accessibility.Certification?, List<MetadataItem>> {
+        var remainingItems = items
+
+        var certification = remainingItems
+            .takeFirstWithProperty(Vocabularies.A11Y + "certifiedBy")
+            .let { remainingItems = it.second; it.first }
+            ?.toCertification()
+            ?: Accessibility.Certification(certifiedBy = null, credential = null, report = null)
+
+        if (certification.credential == null) {
+            remainingItems.takeFirstWithProperty(Vocabularies.A11Y + "certifierCredential")
+                .let { remainingItems = it.second; it.first }
+                ?.let { certification = certification.copy(credential = it.value) }
+        }
+
+        if (certification.report == null) {
+            remainingItems
+                .takeFirstWithProperty(Vocabularies.A11Y + "certifierReport")
+                .let { remainingItems = it.second; it.first }
+                ?.let { certification = certification.copy(report = it.value) }
+        }
+
+        if (certification.report == null) {
+            remainingItems
+                .takeFirstWithRel(Vocabularies.A11Y + "certifierReport")
+                .let { remainingItems = it.second; it.first }
+                ?.let { certification = certification.copy(report = it.href) }
+        }
+
+        return if (remainingItems.size == items.size) {
+            null to remainingItems
+        } else {
+            certification to remainingItems
+        }
+    }
+
+    private fun MetadataItem.Meta.toCertification(): Accessibility.Certification {
+        require(property == Vocabularies.A11Y + "certifiedBy")
+
+        val credential = children
+            .firstWithProperty(Vocabularies.A11Y + "certifierCredential")?.value
+
+        val report = children
+            .firstWithRel(Vocabularies.A11Y + "certifierReport")?.href
+
+        return Accessibility.Certification(
+            certifiedBy = value,
+            credential = credential,
+            report = report
+        )
+    }
+
+    private fun accessibilityProfileFromString(value: String): Accessibility.Profile? = when {
+        isWCAG_20_A(value) -> Accessibility.Profile.EPUB_A11Y_10_WCAG_20_A
+        isWCAG_20_AA(value) -> Accessibility.Profile.EPUB_A11Y_10_WCAG_20_AA
+        isWCAG_20_AAA(value) -> Accessibility.Profile.EPUB_A11Y_WCAG_20_AAA
+        else -> null
+    }
+
+    private fun isWCAG_20_A(value: String) = value in setOf(
+        "EPUB Accessibility 1.1 - WCAG 2.0 Level A",
+        "http://idpf.org/epub/a11y/accessibility-20170105.html#wcag-a",
+        "http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-a",
+        "http://wwwidpf.org/epub/a11y/accessibility-20170105.html#wcag-a",
```

Comment:
Where is this one coming from?

Reply:
That's a weird typo. I meant `https://idpf.org/epub/a11y/accessibility-20170105.html#wcag-a`.